backend/src/controller/defenceController.ts

```diff
@@ -89,23 +96,40 @@ function handleDefenceDeactivation(req: DefenceActivateRequest, res: Response) {
 		return;
 	}
 
-	if (req.session.levelState[3].defences === undefined) return; // this dumb type guard is needed, because we are accessing the array via enum, and so typescript doesn't account for the above type guards. This will be solved with #842. The below array indices won't need to be hardcoded either.
+	const defence = currentDefences.find((defence) => defence.id === defenceId);
+
+	if (defence === undefined) {
+		sendErrorResponse(res, 400, `Defence with id ${defenceId} not found`);
+		return;
+	}
 
-	req.session.levelState[3].defences = deactivateDefence(
+	req.session.levelState[level].defences = deactivateDefence(
 		defenceId,
-		req.session.levelState[3].defences
+		currentDefences
 	);
 	res.status(200).send();
 }
 
 function handleConfigureDefence(req: DefenceConfigureRequest, res: Response) {
-	// id of the defence
-	const defenceId = req.body.defenceId;
-	const config = req.body.config;
-	const level = req.body.level;
+	const { defenceId, level, config } = req.body;
+
+	if (defenceId === undefined) {
+		sendErrorResponse(res, 400, 'Missing defenceId');
+		return;
+	}
+
+	if (level === undefined) {
```

Comment:
`if (!Number.isFinite(level))` will guard against undefined, null or NaN.

Reply:
I've changed them to this `if (!Number.isFinite(level) || level === undefined)` for the sake of type narrowing